src/encoded/batch_download.py

```diff
@@ -216,13 +247,123 @@ def peak_metadata(context, request):
     )
 
 
+def _get_annotation_metadata(request, search_path):
+    """
+    Get annotation metadata.
+
+        :param request: Pyramid's request.
+        :param search_path: Search url.
+    """
+    param_list = parse_qs(request.matchdict['search_params'])
+    param_list['limit'] = ['all']
+    # Handle metadata.tsv lines from cart-generated files.txt.
+    # TODO: When rewriting this file, refactor this duplicated cart code.
+    cart_uuids = param_list.get('cart', [])
+    if cart_uuids:
+        cart_uuid = cart_uuids.pop()
+        del param_list['cart']
+        try:
+            cart = request.embed(cart_uuid, '@@object')
+        except KeyError:
+            pass
+        else:
+            if cart.get('elements'):
+                param_list['@id'] = cart['elements']
+    else:
+        try:
+            elements = request.json.get('elements')
+        except ValueError:
+            pass
+        else:
+            param_list['@id'] = elements
+    path = '{}?{}'.format(search_path, urlencode(param_list, True))
+    results = request.embed(path, as_user=True)
+    annotation_graph = results['@graph']
+    ids = [g['@id'] for g in annotation_graph]
+    file_graph = _get_graph('file', request, ids)
+    software_graph = _get_graph('software', request, ids)
+    headers = [header for header in _tsv_mapping_annotation]
+    fout = io.StringIO()
+    writer = csv.writer(fout, delimiter='\t')
+    writer.writerow(headers)
+    for file in file_graph:
+        annotation = _get_embedded_schema_by_dataset(annotation_graph, file['dataset'])
+        software = _get_embedded_schema_by_dataset(software_graph, file['dataset'])
+        writer.writerow([
+            file.get('title', ''),
+            file.get('file_type', ''),
+            file.get('output_type', ''),
+            annotation.get('accession', ''),
+            annotation.get('annotation_type', ''),
+            software.get('title', ''),
+            annotation.get('encyclopedia_version', ''),
+            annotation.get('biosample_term_id', ''),
+            annotation.get('biosample_term_name', ''),
+            annotation.get('biosample_type', ''),
+            annotation.get('relevant_life_stage', ''),
+            annotation.get('relevant_timepoint', ''),
+            annotation.get('relevant_timepoint_units', ''),
+            annotation.get('organism', {}).get('scientific_name', ''),
+            annotation.get('targets', {}).get('name', ''),
+            annotation.get('date_released', ''),
+            annotation.get('award').get('project', ''),
+            file.get('derived_from', ''),
+            file.get('file_size', ''),
+            file.get('lab', {}).get('title', ''),
+            file.get('md5sum', ''),
+            file.get('dbxrefs', ''),
+            file.get('href', ''),
+            file.get('assembly', ''),
+            file.get('controlled_by', ''),
+            file.get('status', ''),
+            file.get('restricted', ''),
+        ])
```

Comment:
you don't do the audit mapping here?  
Also, I think you also need to add
`                    if prop in ['files.paired_with', 'files.derived_from']:
                        # chopping of path to just accession
                        if len(temp):
                            new_values = [t[7:-1] for t in temp]
                            temp = new_values
`
I think because we have file.derived_from in the list of fields.

Reply:
Opps. Added audit and other columns.